docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,601 @@
+
+3. Create a new file in VS Code (**File** > **New Text File** or `kb(workbench.action.files.newUntitledFile)`).
+4. Add the following content to it:
+    ```
+    fastapi
+    redis
+    types-redis
+    uvicorn
+    ```
+5.	Save the file (`kb(workbench.action.files.save)`) and name it requirements.txt
+6.	Create a virtual environment by opening the Command Palette (`kb(workbench.action.showCommands)`) and running the **Python: Create Environment** command.
+    > Note: this step may take a couple of minutes to complete.
+7. When asked by the environment type, select **Venv**:
+![Drop down with "Venv" or "Conda" as options for environments that can be created with the Python: Create Environment command.](images/environments/create_environment_dropdown.png)
+
+8. Then select the latest version of Python available on your machine:
+![List of available global environments that can be used to create a virtual environment.](images/fastapi-tutorial/create_environment_interpreters_list.png)
+
+9. Select the requirements.txt file from the drop-down list, so the dependencies are automatically installed, and then select "OK":
+![Check box selected to install dependencies from requirements.txt file](images/fastapi-tutorial/create_environment_select_requirements.png)
+
+The virtual environment will be created, the dependencies will be automatically installed and the environment will be selected for your workspace to be used by the Python extension.
+
+## Start Coding
+
+Let’s create the application!
+1. Create a new Python file by clicking on **File** > **New File…** (`kb(workbench.action.files.newFile)`), and then select "Python File".
+2. Save it as `main.py` (`kb(workbench.action.files.saveAs)`) in the `groceries-plugin` folder.
+3. Add the following code to `main.py` and save the file:
+    ```python
+    from fastapi import FastAPI
+
+    app = FastAPI()
+
+    @app.get("/")
+    def root():
+        return {"message": "Hello World"}
+    ```
+4. Run the code by starting up the debugger (`kb(workbench.action.debug.start)`)
+5. From the drop-down menu, select the **FastAPI** configuration option from the drop-down list:
+![Drop down with debugger configuration options, with FastAPI being highlighted.](images/fastapi-tutorial/fastapi_debug_config_option.png)
+
+    This will invoke uvicorn to start the application server through the debugger, which will allow us to step through the code to inspect its behavior. You will see something like the following in the terminal:
+
+    ![Uvicorn server running message displayed in the terminal, with an URL to access the app.](images/fastapi-tutorial/fastapi_debug_terminal.png)
+
+6. `Ctrl` + click the http://127.0.0.1:8000/ URL in the terminal to open your default browser to that address:
+![Hello World message displayed in the browser.](images/fastapi-tutorial/helloworld_browser.png)
+7. Stop the debugger by clicking on the stop icon in the debug toolbar, or through `kb(workbench.action.debug.stop)`.
```

Comment:
Could we between 6 and 7 say something like "congrats your FastAPI is running! You can set a breakpoint..."
Or something like that. I would like a sense of achievement here :)

Reply:
I love that! This inspired me to add a few more congrats on success points, as well as a thank you note at the end. Thanks for calling that out 😊